Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -0,0 +1,1203 @@
+                }
+            }
+
+            bool isExternal = haveSource && !IsInLocalNetwork(sourceAddr);
+
+            string bindPreference = string.Empty;
+            if (haveSource)
+            {
+                // Check for user override.
+                foreach (var addr in _overrideUrls)
+                {
+                    // Remaining. Match anything.
+                    if (addr.Key.Equals(IPAddress.Broadcast))
+                    {
+                        bindPreference = addr.Value;
+                        break;
+                    }
+                    else if ((addr.Key.Equals(IPAddress.Any) || addr.Key.Equals(IPAddress.IPv6Any)) && (isExternal || chromeCast))
+                    {
+                        // External.
+                        bindPreference = addr.Value;
+                        break;
+                    }
+                    else if (addr.Key.Contains(sourceAddr))
+                    {
+                        // Match ip address.
+                        bindPreference = addr.Value;
+                        break;
+                    }
+                }
+            }
+
+            _logger.LogDebug("GetBindInterface: Souce: {0}, External: {1}:", haveSource, isExternal);
+
+            if (!string.IsNullOrEmpty(bindPreference))
+            {
+                // Has it got a port defined?
+                var parts = bindPreference.Split(':');
+                if (parts.Length > 1)
+                {
+                    if (int.TryParse(parts[1], out int p))
+                    {
+                        bindPreference = parts[0];
+                        port = p;
+                    }
+                }
+
+                _logger.LogInformation("{0}: Using BindAddress {1}:{2}", sourceAddr, bindPreference, port);
+                return bindPreference;
+            }
+
+            string ipresult;
+
+            // No preference given, so move on to bind addresses.
+            lock (_intLock)
+            {
+                var nc = _bindAddresses.Exclude(_bindExclusions).Where(p => !p.IsLoopback());
+
+                int count = nc.Count();
+                if (count == 1 && (_bindAddresses[0].Equals(IPAddress.Any) || _bindAddresses.Equals(IPAddress.IPv6Any)))
+                {
+                    // Ignore IPAny addresses.
+                    count = 0;
+                }
+
+                if (count != 0)
+                {
+                    // Check to see if any of the bind interfaces are in the same subnet.
+
+                    IEnumerable<IPObject> bindResult;
+                    IPAddress? defaultGateway = null;
+
+                    if (isExternal)
+                    {
+                        // Find all external bind addresses. Store the default gateway, but check to see if there is a better match first.
+                        bindResult = nc.Where(p => !IsInLocalNetwork(p)).OrderBy(p => p.Tag);
```

Comment:
`bindResult` is enumerated multiple times

Reply:
altered